test/e2e/e2e_test.go

```diff
@@ -134,83 +133,18 @@ func namespaceForTest(t *testing.T) string {
 
 // deployBroker will deploy the broker inside the specified namespace.
 func deployBroker(namespace string) error {
-	numOfReplicas := int32(1)
+	deploy := &appsv1.Deployment{}
+	testutil.ReadInYAMLFileAndConvert("../../samples/kubernetes/used_for_e2e_tests/deploy.yaml", &deploy)
+	_, err := kubeClient.AppsV1().Deployments(namespace).Create(deploy)
 
-	// Create deployment of the broker server.
-	_, err := kubeClient.AppsV1().Deployments(namespace).Create(&appsv1.Deployment{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: name,
-		},
-		Spec: appsv1.DeploymentSpec{
-			Replicas: &numOfReplicas,
-			Selector: &metav1.LabelSelector{
-				MatchLabels: map[string]string{
-					"app": name,
-				},
-			},
-			Template: v1.PodTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{
-					Labels: map[string]string{
-						"app": name,
-					},
-				},
-				Spec: v1.PodSpec{
-					Containers: []v1.Container{
-						v1.Container{
-							Name:            name,
-							Image:           image,
-							ImagePullPolicy: "Never",
-							Ports: []v1.ContainerPort{
-								v1.ContainerPort{
-									ContainerPort: int32(4000),
-								},
-							},
-							Env: []v1.EnvVar{
-								v1.EnvVar{
-									Name:  "ATLAS_BASE_URL",
-									Value: atlasBaseURL,
```

Comment:
this env variable is not in the yaml file - is this intentional?

Reply:
Yikes, Will add it. Otherwise I'll just maybe find a way to improve it.